src/log.c

```diff
@@ -133,6 +136,58 @@ void fi_log_init(void)
 	pid = getpid();
 }
 
+static int ofi_log_enabled(const char *prov_name,
+			   enum fi_log_level level, enum fi_log_subsys subsys,
+			   uint64_t flags)
+{
+	int prov_filtered = (flags & FI_LOG_PROV_FILTERED);
+
+	return ((FI_LOG_TAG(prov_filtered, level, subsys) & log_mask) ==
+		FI_LOG_TAG(prov_filtered, level, subsys));
+}
+
+static void ofi_log(const char *prov_name, enum fi_log_level level,
+		    enum fi_log_subsys subsys, const char *func, int line,
+		    const char *msg)
+{
+	fprintf(stderr, "%s:%d:%ld:%s:%s:%s:%s():%d<%s> %s",
+		PACKAGE, pid, (unsigned long) time(NULL), log_prefix,
+		prov_name, log_subsys[subsys], func, line,
+		log_levels[level], msg);
+}
+
+static int ofi_log_ready(const char *prov_name, enum fi_log_level level,
+			 enum fi_log_subsys subsys, uint64_t flags,
+			 uint64_t *showtime);
```

Comment:
Why is the implementation for this below?

Reply:
There is a cyclic dependency problem here: 
ofi_log_ready() -> log_fid -> ofi_import_log_ops -> ofi_bind_logging_fid() -> ofi_logging_import() -> log_fid

I will try to break the cyclic dependency.